src/neuroconv/tools/nwb_helpers/_configure_backend.py

```diff
@@ -21,4 +22,14 @@ def configure_backend(
 
         # TODO: update buffer shape in iterator, if present
 
-        nwbfile_objects[object_id].set_data_io(dataset_name=dataset_name, data_io_class=data_io_class, **data_io_kwargs)
+        if isinstance(nwbfile_objects[object_id], Data):
```

Comment:
I have make this point before but I don't remember you answer, why do you favor in-lining definitions so much? 

It does get in the way of my debugging workflow. That's not a strong argument but if there is no argument in favor either could we do something like this instead?
```
        # TODO: update buffer shape in iterator, if present
        neurodata_object = nwbfile_objects[object_id]
        if isinstance(neurodata_object, Data):
            neurodata_object.set_data_io(data_io_class=data_io_class, data_io_kwargs=data_io_kwargs)
        elif isinstance(neurodata_object, TimeSeries):
            neurodata_object.set_data_io(
                dataset_name=dataset_name, data_io_class=data_io_class, **data_io_kwargs
            )
        else:  # Strictly speaking, it would be odd if a backend_configuration led to this, but might as well be safe
            raise NotImplementedError(
                f"Unsupported object type {type(neurodata_object)} for backend "
                f"configuration of {neurodata_object.name}!"
            )
```

Reply:
It simply does not interfere with my debugging workflow, nor does it alter readability from my perspective when it's something so small (literally just fetching a value from a dict in this case)

I'm fine with doing it your way if you comment on it, but in general IDK if I will ever make it as hyper focused of a habit as you do